### ISSUE TITLE
Warning about "createMachine" with @aws-amplify/ui-react when running tests

### ISSUE THREAD

**Issue by sheelah** [aws-amplify/amplify-ui#2732]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Authenticator

### How is your app built?

Vite

### What browsers are you seeing the problem on?

Chrome

### Please describe your bug.

We're seeing this error coming up when running our tests, which comes from our use of the `@aws-amplify/ui-react@3.4.2` package:

```
It is highly recommended to set `predictableActionArguments` to `true` when using `createMachine`. https://xstate.js.org/docs/guides/actions.html
```

We have components that rely on the `useAuthenticator` hook from Amplify. We use React testing library and Vitest for tests and currently mock the `useAuthenticator` hook with:

```
vi.spyOn(useAuthenticatorModule, 'useAuthenticator');
```

Is there a recommended approach for mocking this hook for testing purposes? The mocking approach we have in place works but we haven't been able to get rid of these errors.

### What's the expected behaviour?

No errors relating to Amplify show up in test runs.

### Help us reproduce the bug!

In your tests, render a component that calls the `useAuthenticator` hook as a child of  `<Authenticator.Provider>` like this:

```
<Authenticator.Provider>
 <ChildComponent />
</Authenticator.Provider>
```

### Code Snippet

```typescript
// Helper for rendering components that rely on `Authenticator.Provider` React context:
export const renderWithAmplify = (ui: React.ReactElement) => {
  Amplify.configure(AMPLIFY_CONFIG);
  const Wrapper = ({children}: {children: React.ReactNode}) => (
    <Authenticator.Provider>{children}</Authenticator.Provider>
  );

  return {...render(ui, {wrapper: Wrapper})};
};


// UseAuthenticator spy
const useAuthenticatorSpy = vi.spyOn(useAuthenticatorModule, 'useAuthenticator');

// Both the Authenticator context helper and spy used in a test
test('renders xyz if the user is logged in',  async () => {
    useAuthenticatorSpy.mockReturnValue({route: 'authenticated'} as any);

    renderWithAmplify(
      <MyChildComponent />
    );
})
```

### Additional information and screenshots

![Screen Shot 2022-10-12 at 1 18 59 PM](https://user-images.githubusercontent.com/1900318/195429382-ab20115c-7b5d-4ecc-9263-4b6969684abb.png)

**Comment by wlee221** [comment]
Hi @sheelah! This error should be gone after https://github.com/aws-amplify/amplify-ui/pull/2569, which was released in `@aws-amplify/ui-react@3.5.2` ([release notes](https://github.com/aws-amplify/amplify-ui/pull/2578)). Can you try upgrading to the that (or preferably latest) version, and let us know if that removes the warning?

**Comment by sheelah** [comment]
This worked - thank you!
